Detections/OfficeActivity/Mail_redirect_via_ExO_transport_rule.yaml

```diff
@@ -0,0 +1,49 @@
+id: 500415fb-bba7-4227-a08a-9857fb61b6a7
+name: Mail redirect via ExO transport rule
+description: |
+  'Identifies when Exchange Online transport rule configured to forward emails.
+  This could be an adversary mailbox configured to collect mail from multiple user accounts.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Office365
+    dataTypes:
+      - OfficeActivity
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Collection
+  - Exfiltration
+relevantTechniques:
+  - T1114
+  - T1020
+query: |
+
+  let timeframe = 1d;
+  OfficeActivity
+  | where TimeGenerated >= ago(timeframe)
+  | where OfficeWorkload == "Exchange"
+  | where Operation in~ ("New-TransportRule", "Set-TransportRule")
+  | extend p = parse_json(Parameters)
+  | extend RuleName = case(
+    Operation =~ "Set-TransportRule", tostring(OfficeObjectId),
+    Operation =~ "New-TransportRule", tostring(p[1].Value),
+    "Unknown"
+    ) 
+  | mvexpand p
+  | where (p["Name"] == "BlindCopyTo" and isnotempty(p["Value"])) or (p["Name"] == "RedirectMessageTo" and isnotempty(p["Value"]))
+  | extend RedirectTo = p["Value"]
+  | extend ClientIPOnly = case( 
+    ClientIP has ".", tostring(split(ClientIP,":")[0]), 
```

Comment:
I think there is another case to handle here, which is this format - 10.10.10.10-9999.  I have seen this occur in New-TransportRule.  Maybe this would be better?

  | extend ClientIPonly = trim_end('[:\\-\\[].*', ClientIP)
  | extend Port = trim_end('\\]',trim_start('[0-9]*\\.[0-9]*\\.[0-9]*\\.[0-9]*[:\\-\\[]*', ClientIP))

Reply:
@shainw , I just verified your code. It has a syntax error.

`let ClientIP = "[2001:67c:2660:425:21b:21ff:feb1:5c80]:31290";
print trim_end('[:\-\[].', ClientIP)

let ClientIP = "10.10.10.10:9000";
print trim_end('[:\-\[].', ClientIP)`

I will add support for another format of ClientIP.

Thank you for advice!

Reply:
Yeah, sorry on that...the web formatting of Github removed my escapes since I forgot to enclose it in a code block surrounded by ``.  :)  I did come up with something that may also work, but your changes are good.  

`| extend IP = case(
ClientIP startswith "[", trim_end("[:\\-][0-9\\[0-9]*]*", ClientIP),
trim_end("[:\\-][0-9\\[0-9]*]*", ClientIP)
)
| extend Port = case(ClientIP startswith "[",trim_end('\\]',trim_start('\\[[a-zA-Z0-9:]*[:\\[\\]\\-]*', ClientIP)), 
trim_end('\\]',trim_start('[0-9\\.]*[:\\[\\-]*', ClientIP))
)`